### PR TITLE
Gb 4992 send application/json content type by default

### PR DESCRIPTION
**Pull request by jonathanawesome**
This PR offers two enhancements.

First, it adds the `content-type: application/json` header to the initial headers state. Every newly initialized session will have this header added.

Second, it adds the ability to add multiple headers on the Connect screen, should the user require them.

### CODE REVIEW

**Comment by fbjork** [comment]
@jonathanawesome nice. The content type header should not be added to the list since it’s a default value.

File: packages/stores/src/session-store/slices/http-headers/http-headers-state.ts [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
 import { HTTPHeadersState } from "./http-headers.types";
 
-const INITIAL_HTTP_HEADERS_STATE: HTTPHeadersState = {
-  headers: [],
+export const INITIAL_HTTP_HEADERS_STATE: HTTPHeadersState = {
+  headers: [
+    {
+      id: generateCuid({}),
+      enabled: true,
+      key: "content-type",
```
Comment (yoav-lavi) [inline packages/stores/src/session-store/slices/http-headers/http-headers-state.ts:9]:
```suggestion
      key: "Content-Type",
```

File: packages/core/src/components/connect/connect.tsx [hunk 2]
```diff
@@ -46,21 +87,26 @@ export const Connect = () => {
   const introspectionErrors = useSchemaStore.use.introspectionErrors();
 
   const [endpointValue, setEndpointValue] = useState<string>("");
-  const [headerKeyValue, setHeaderKeyValue] = useState<string>("");
-  const [headerValueValue, setHeaderValueValue] = useState<string>("");
+  const [headers, headersDispatch] = useReducer(headersReducer, [
+    {
+      id: generateCuid({}),
+      enabled: true,
+      key: "content-type",
```
Comment (yoav-lavi) [inline packages/core/src/components/connect/connect.tsx:94]:
```suggestion
      key: "Content-Type",
```

**Comment by yoav-lavi** [comment]
If you add too many headers this doesn't scroll, I'd allow scrolling IMO

**Comment by yoav-lavi** [comment]
@jonathanawesome the header changes are since some http1 servers will be cased even though it's against the standard, I think even we have some instances of this behavior. http2 will just lowercase so this supports the most cases.


We should probably add `User-Agent: pathfinder/version` as well

**Comment by yoav-lavi** [comment]
image

**Review by yoav-lavi — APPROVED** [review]

**Pull request merged**